src/frontend/src/forms/PurchaseOrderForms.tsx

```diff
@@ -66,3 +62,52 @@ export function purchaseOrderLineItemFields({
 
   return fields;
 }
+
+/**
+ * Construct a set of fields for creating / editing a PurchaseOrder instance
+ */
+export function purchaseOrderFields(): ApiFormFieldSet {
+  return {
+    reference: {
+      icon: <IconHash />
+    },
+    description: {},
+    supplier: {
+      filters: {
+        is_supplier: true
+      }
+    },
+    supplier_reference: {},
+    project_code: {
+      icon: <IconList />
+    },
+    order_currency: {
+      icon: <IconCoins />
+    },
+    target_date: {
+      icon: <IconCalendar />
+    },
+    link: {},
+    contact: {
+      icon: <IconUser />,
+      adjustFilters: (value: ApiFormAdjustFilterType) => {
```

Comment:
@wolflu05 see how much simpler it is now to adjust the filters of one field based on the value of another field?